trivup/apps/SchemaRegistryApp.py

```diff
@@ -28,96 +28,115 @@
+                                                         self.default_version),
+                                                conf)
+
+        protocol = 'http'
+        # Find available port on the host and expose it
+        port = trivup.TcpPortAllocator(cluster).next(
+            self, self.conf.get('port_base', None))
+        self.expose_port(self.docker_args, 8081, port)
 
-        if self.conf.get('image', '') == '':
-            self.conf['image'] = self.default_image
+        app_conf = dict(self.default_app_conf, **conf.get('conf', {}))
 
-        self.conf['container_id'] = 'trivup_sr_%s' % str(uuid.uuid4())[0:7]
         kafka = cluster.find_app(KafkaBrokerApp)
-        if kafka is None:
+        if kafka:
+            app_conf['kafkastore.bootstrap.servers'] =\
+                kafka.conf.get('docker_advertised_listeners')
+        else:
             raise Exception('KafkaBrokerApp required')
 
-        bootstrap_servers = kafka.conf.get('docker_advertised_listeners')
-
-        if bootstrap_servers is None:
-            raise Exception('KafkaBrokerApp required')
-
-        # Create listener
-        port = trivup.TcpPortAllocator(self.cluster).next(
-            self, self.conf.get('port_base', None))
-
-        docker_args = ''
-        if cluster.platform == 'linux':
-            # Let container bind to host localhost
-            self.conf['extport'] = port
-            self.conf['intport'] = port
-            docker_args = '--network=host'
-
-        elif cluster.platform == 'darwin':
-            # On OSX localhost binds are not possible, so set up a
-            # port forwarding.
-            self.conf['extport'] = port
-            self.conf['intport'] = 8081
-            docker_args = '-p %d:%d' % (self.conf['extport'],
-                                        self.conf['intport'])
-
-        # This is the listener address inside the docker container
-        self.conf['listeners'] = 'http://0.0.0.0:%d' % self.conf.get('intport')
-        # This is the listener address outside the docker container,
-        # using port-forwarding
-        self.conf['url'] = 'http://localhost:%d' % self.conf['extport']
-
-        # Run in foreground.
-        self.conf['start_cmd'] = 'docker run %s --name %s -e SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS=%s  -e SCHEMA_REGISTRY_HOST_NAME=localhost   -e SCHEMA_REGISTRY_LISTENERS=%s  -e SCHEMA_REGISTRY_DEBUG=true %s' % (  # noqa: E501
-            docker_args,
-            self.conf.get('container_id'),
-            bootstrap_servers,
-            self.conf.get('listeners'),
-            self.conf.get('image'))
-
-        # Stop through docker
-        self.conf['stop_cmd'] = 'docker stop %s' % \
-                                self.conf.get('container_id')
+        ssl = cluster.find_app(SslApp)
+        if ssl is not None:
+            # Mount SslApp dir, reuse broker trusts/certs
+            self.add_mount(self.docker_args, "/opt/ssl", ssl.root_path())
```

Comment:
/opt may interfer with what is already on the image, use something unique instead, e.g., /trivup/ssl